jepsen/scalardb/src/scalardb/transfer_append.clj

```diff
@@ -0,0 +1,326 @@
+                    (try-reconnection! test)
+                    (assoc op :type :fail, :error "Skipped due to no connection")))
+      :get-all  (if-let [results (get-balances-and-ages-and-versions test (:num op))]
+                  (assoc op :type :ok :value results)
+                  (assoc op :type :fail, :error "Failed to get balances and ages"))
+      :check-tx (let [unknown (:unknown-tx test)]
+                  (if-let [num-committed (if (empty? @unknown)
+                                           0
+                                           (scalar/check-transaction-states test @unknown))]
+                    (assoc op :type :ok, :value num-committed)
+                    (assoc op :type :fail, :error "Failed to check status")))))
+
+  (close! [_ _])
+
+  (teardown! [_ test]
+    (scalar/close-all! test)))
+
+(defn transfer
+  [test _]
+  (let [n (-> test :model :num)]
+    {:type :invoke
+     :f :transfer
+     :value {:from (rand-int n)
+             :to (rand-int n)
+             :amount (+ 1 (rand-int 1000))}}))
+
+(def diff-transfer
+  (gen/filter (fn [op] (not= (-> op :value :from)
+                             (-> op :value :to)))
+              transfer))
+
+(defn get-all
+  [test _]
+  {:type :invoke
+   :f :get-all
+   :num (-> test :model :num)})
+
+(defn check-tx
+  [test _]
+  {:type :invoke
+   :f :check-tx})
+
+(defn consistency-checker
+  []
+  (reify checker/Checker
+    (check [this test history opts]
+      (let [read-result (->> history
+                             (r/filter op/ok?)
+                             (r/filter #(= :get-all (:f %)))
+                             (into [])
+                             last
+                             :value)
+            actual-balance (->> read-result
+                                (map :balance)
+                                (reduce +))
+            bad-balance (if-not (= actual-balance total-balance)
+                          {:type     :wrong-balance
+                           :expected total-balance
+                           :actual   actual-balance})
+            actual-age (->> read-result
+                            (map :age)
+                            (reduce +))
+            ; every ok transfer will increase both parties ages by 1
+            expected-age (->> history
+                              (r/filter op/ok?)
+                              (r/filter #(= :transfer (:f %)))
+                              (into [])
+                              count
+                              (* 2))
+            bad-age (if-not (= actual-age expected-age)
+                      {:type :wrong-age
+                       :expected expected-age
+                       :actual actual-age})
+            actual-version (->> read-result
+                                (map :version)
+                                (reduce +))
+            expected-version (-> NUM_ACCOUNTS)
+            bad-version (if-not (= actual-version expected-version)
```

Comment:
As I made a comment above, I think this version checking isn't needed.
Instead of this checking, can you check the number of records?
I think it needs to scan records by each account ID after transfers.

Reply:
Right! So now it is scanning all the ids, adding up the number of records, and comparing to the ages extracted from the records. Thank you.